build.proj

```diff
@@ -146,6 +150,15 @@
     <MSBuild Projects="@(FunctionalTests)" Properties="TestTargetOS=$(TestOS)netcoreapp;$(TestProjectProperties);Platform=$(Platform);" Condition="$(ReferenceType.Contains('Package'))"/>
   </Target>
 
+  <Target Name="BuildUnitTestsNetCore" DependsOnTargets="RestoreTestsNetCore">
+    <Message Text=">>> Building UnitTestsNetCore [TestTargetOS=$(TestOS)netcoreapp;$(TestProjectProperties);Platform=AnyCPU;ReferenceType=$(ReferenceType);] ..." Condition="!$(ReferenceType.Contains('Package'))"/>
+    <MSBuild Projects="@(UnitTests)" Properties="TestTargetOS=$(TestOS)netcoreapp;$(TestProjectProperties);Platform=AnyCPU;" Condition="!$(ReferenceType.Contains('Package'))" />
+
+    <!-- Only build platform specific builds for Package reference types -->
+    <Message Text=">>> Building FunctionalTestsNetCore [TestTargetOS=$(TestOS)netcoreapp;$(TestProjectProperties);Platform=$(Platform);ReferenceType=$(ReferenceType);] ..." Condition="$(ReferenceType.Contains('Package'))"/>
```

Comment:
Did you want to make this for UnitTests project too?

Reply:
@cheenamalhotra  Yeah, using package reference is not the goal.

 Unit tests are typically executed for tight loop development, and are tightly coupled with the internals of the source code. So expecting that the UT will run with a referenced package doesn't seem right, since there needs to be a good tracking between Unit tests and the packages that they can work with. 

On the other hand the FunctionalTests and Manual Tests use public APIs of the driver, so I expect those tests to work well with the package reference.